printer.go

```diff
@@ -136,6 +173,135 @@ type DriverInfo struct {
 	Attributes  uint32
 }
 
+// PrintJobInfo stores information about a print job.
+type PrintJobInfo struct {
+	JobID           uint32
+	UserMachineName string
+	UserName        string
+	DocumentName    string
+	DataType        string
+	Status          string
+	StatusCode      uint32
+	Priority        uint32
+	Position        uint32
+	TotalPages      uint32
+	PagesPrinted    uint32
+	PercentComplete float32
+	Submitted       time.Time
+}
+
+// PrintJobs returns information about all print jobs on this printer
+func (p *Printer) PrintJobs() ([]PrintJobInfo, error) {
+	var bytesNeeded, jobsReturned uint32
+	var err error
+	buf := make([]byte, 1)
+	for {
+		err = EnumJobs(p.h, 0, 255, 1, &buf[0], uint32(len(buf)), &bytesNeeded, &jobsReturned)
+		if err == nil {
+			break
+		}
+		if err != syscall.ERROR_INSUFFICIENT_BUFFER {
+			return []PrintJobInfo{}, err
+		}
+		if bytesNeeded <= uint32(len(buf)) {
+			return []PrintJobInfo{}, err
+		}
+		buf = make([]byte, bytesNeeded)
+	}
+	if jobsReturned > 0 {
+		pj := make([]PrintJobInfo, jobsReturned)
+		ji := make([]JOB_INFO_1, jobsReturned)
```

Comment:
That does work nicely and removes the need for the reflect package but this could then cause a "slice bounds out of range" runtime error if jobsReturned > 1024. We could cap jobsReturned prior to doing this if we don't want to worry about it. Modifying the slice header with reflect did allow the number of jobs returned to be what ever was returned though.

Reply:
You can replace 1024 with any const you like - whatever you feel comfortable. Make it 1024 * 1024 if you like. Or even bigger. Lets not check for jobsReturned value before using it.